### ISSUE TITLE
Feature Requests

### ISSUE THREAD

**Issue by RobinCGN** [C-C-A-A/CologneProtocol-R#19]
@VicluUiB requested a couple of features:

- [ ] Export of plots as images

- [ ] Smoothing of Isolines similar to QGIS/MapInfo results ("_You Isolines look like LEGO!_")

**Comment by RobinCGN** [comment]
1) I have added an experimental image export function (plot to *.svg, raster maps to *.tif) to _03_Visualisation_Export.R_.


2) Smoothing of isolines can be achieved using the **smoothr** package:

`require(sp)
library(smoothr)

'# convert polygons to lines
isoline_merged_lines <- as(isoline_merged, 'SpatialLinesDataFrame')

'# smooth isolines or polygons (smoothness can be adjusted)
iso_merged_lines_smo <- smooth(isoline_merged_lines, method = "ksmooth", smoothness = 2)
iso_merged_poly_smo <- smooth(isoline_merged, method = "ksmooth", smoothness = 2)

'# export
rgdal::writeOGR(iso_merged_lines_smo,
                dsn = "playground",
                layer = "iso_merged_lines_smo",
                driver = "ESRI Shapefile",
                check_exists = TRUE,
                overwrite_layer = TRUE)

rgdal::writeOGR(iso_merged_poly_smo,
                dsn = "playground",
                layer = "iso_merged_poly_smo",
                driver = "ESRI Shapefile",
                check_exists = TRUE,
                overwrite_layer = TRUE)`

**Comment by Mono2** [comment]
I used `smoothr` in a previous version of my local script. In the end, however, I decided not to smooth the isolines, although it looks "nicer". The difference in the statistical values is considerable.
My suggestion would be that the calculation of the statistical values is always based on the non-smoothed isolines and that there is the **option** of having smoothed isolines on the images or in the export of the shape-filese.

**Comment by RobinCGN** [comment]
You're absolutely right, smoothed isolines shouldn't be used for any kind of calculations. Now that you agree, will add this feature as option in the next days, but I don't know how the package dependencies stuff (deps.yaml) works.

**Comment by Mono2** [comment]
No Problem. I will update it when you are finished.

**Comment by VicluUiB** [comment]
Thank you both for keeping me updated with all the fantastic work that you're doing!
As far as smoothing of the isolines go, I agree. However, it would be a great function to have as it would make more sense to illustrate the outputs in this way when making figures for e.g. publication or other types of dissemination.

**Comment by RobinCGN** [comment]
I have added the smoothing of isolines to the regular script. I made it optional or rather dependent on "merge polygons". If the "merge-polygons"-option is "TRUE", a .shp with smoothed isolines is produced.
